File: src/integrations/notion.ts
import { Client } from "@notionhq/client";

import { config } from "../config.js";

export function getNotion(): Client {
  if (!config.NOTION_TOKEN) throw new Error("NOTION_TOKEN not configured");
  return new Client({ auth: config.NOTION_TOKEN, notionVersion: "2022-06-28" });
}

File: src/integrations/notion.py
from notion_client import AsyncClient

from src.config import get_settings
from src.exceptions import ConfigError


def get_notion() -> AsyncClient:
    settings = get_settings()
    if not settings.NOTION_TOKEN:
        raise ConfigError("NOTION_TOKEN not configured")
    return AsyncClient(auth=settings.NOTION_TOKEN, notion_version="2022-06-28")
